tools/c7n_azure/c7n_azure/actions/tagging.py

```diff
@@ -281,21 +287,27 @@ def _get_tag_value_from_event(self, event):
         elif StringUtils.equal(principal_type, 'ServicePrincipal'):
             user = self.sp_jmes_path.search(event)
 
-        # Other types and main fallback (e.g. User, Office 365 Groups, and Security Groups)
-        if not user and self.upn_jmes_path.search(event):
-            user = self.upn_jmes_path.search(event)
+        if not user:
+            known_claims = {'upn': self.upn_jmes_path.search(event),
+                            'name': self.name_jmes_path.search(event)}
+            if known_claims[self.default_claim]:
+                user = known_claims[self.default_claim]
+            elif self.default_claim == 'upn' and known_claims['name']:
```

Comment:
Is it not sufficient to just set the user = known_claims['name'] or user = known_claims['upn'] if values exist in `known_claims`?

Does it matter what the default_claim is?

Reply:
They can exist at the same time.. I'm not 100% sure in what case they might be different, but I'd expect it to be the case.